Git town crashes trying to parse some git symbolic ref 's

### Steps to reproduce

```
brew install git-town
cd existing-git-dir
git symbolic-ref refs/remotes/origin/main refs/remotes/origin/master
git town hack 1-test
```

Most git town commands fail afterwards.  Deleting ^ the symbolic link fixes it.  Other symbolic links do not cause issues e.g. `git symbolic-ref refs/heads/main refs/heads/master`.

### Command output

```shell
$ git town hack 2-test                                                
panic: "->" is not a valid Git SHA

goroutine 1 [running]:
github.com/git-town/git-town/v18/internal/git/gitdomain.NewSHA(...)
        github.com/git-town/git-town/v18/internal/git/gitdomain/sha.go:16
github.com/git-town/git-town/v18/internal/git.ParseVerboseBranchesOutput({0x14000a60000, 0x27b3a5})
        github.com/git-town/git-town/v18/internal/git/commands.go:992 +0xa64
github.com/git-town/git-town/v18/internal/git.(*Commands).BranchesSnapshot(0x1400023ca48, {0x105a51690, 0x140000ea7c8})
        github.com/git-town/git-town/v18/internal/git/commands.go:100 +0x1d0
github.com/git-town/git-town/v18/internal/cmd.determineHackData({_, _, _}, {{0x105a530b0, 0x140000ea7c8}, {0x140000bb1f8}, {0x1400017d7a0, 0x1400017d9b0}, {0x140000ea798}, {0x105a4f980, ...}, ...}, ...)
        github.com/git-town/git-town/v18/internal/cmd/hack.go:186 +0xf4
github.com/git-town/git-town/v18/internal/cmd.executeHack({0x140000a8b30, 0x1, 0x1}, 0x0, 0x0, 0x0, 0x0)
        github.com/git-town/git-town/v18/internal/cmd/hack.go:91 +0x16c
github.com/git-town/git-town/v18/internal/cmd.hackCmd.func1(0x14000211208, {0x140000a8b30, 0x1, 0x1})
        github.com/git-town/git-town/v18/internal/cmd/hack.go:69 +0xd0
github.com/spf13/cobra.(*Command).execute(0x14000211208, {0x140000a8b00, 0x1, 0x1})
        github.com/spf13/cobra@v1.8.1/command.go:985 +0x834
github.com/spf13/cobra.(*Command).ExecuteC(0x140000c6608)
        github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/git-town/git-town/v18/internal/cmd.Execute()
        github.com/git-town/git-town/v18/internal/cmd/doc.go:45 +0x3d0
main.main()
        github.com/git-town/git-town/v18/main.go:18 +0x28
```

--verbose output is too long due to large repo size and might yield sensitive info, sorry.

### Additional information

_No response_

--- a/internal/git/commands.go
+++ b/internal/git/commands.go
@@ -930,7 +930,11 @@ func ParseVerboseBranchesOutput(output string) (gitdomain.BranchInfos, Option[gi
 			sha = Some(gitdomain.NewSHA(parts[4]))
 		} else {
 			branchName = parts[0]
-			sha = Some(gitdomain.NewSHA(parts[1]))
+			rawSHA := parts[1]
+			if rawSHA == "->" {
+				continue
+			}
+			sha = Some(gitdomain.NewSHA(rawSHA))
 		}
 		remoteText := parts[2]
 		if line[0] == '*' {

--- a/internal/git/commands_test.go
+++ b/internal/git/commands_test.go
@@ -862,6 +862,28 @@ func TestBackendCommands(t *testing.T) {
 			must.Eq(t, want, have)
 		})
 
+		t.Run("symbolic reference", func(t *testing.T) {
+			t.Parallel()
+			give := `
+  main                                   4dc97db26 [origin/main] Commit 1
+  remotes/origin/HEAD                    -> origin/main
+  remotes/origin/main                    4dc97db26 Commit 1
+  remotes/origin/master                  -> origin/main
+`[1:]
+			want := gitdomain.BranchInfos{
+				gitdomain.BranchInfo{
+					LocalName:  Some(gitdomain.NewLocalBranchName("main")),
+					LocalSHA:   Some(gitdomain.NewSHA("4dc97db26")),
+					SyncStatus: gitdomain.SyncStatusUpToDate,
+					RemoteName: Some(gitdomain.NewRemoteBranchName("origin/main")),
+					RemoteSHA:  Some(gitdomain.NewSHA("4dc97db26")),
+				},
+			}
+			have, currentBranch := git.ParseVerboseBranchesOutput(give)
+			must.Eq(t, want, have)
+			must.Eq(t, None[gitdomain.LocalBranchName](), currentBranch)
+		})
+
 		t.Run("complex example", func(t *testing.T) {
 			give := `
   branch-1                     01a7eded [origin/branch-1: ahead 1] Commit message 1a

